support/cas-server-support-oidc-core-api/src/main/java/org/apereo/cas/oidc/web/controllers/logout/OidcLogoutEndpointController.java

```diff
@@ -32,8 +37,16 @@
  */
 @Slf4j
 public class OidcLogoutEndpointController extends BaseOAuth20Controller {
-    public OidcLogoutEndpointController(final OidcConfigurationContext context) {
+
+    private final UrlValidator urlValidator;
+
+    private final OidcPostLogoutRedirectUrlMatcher postLogoutRedirectUrlMatcher;
+
+    public OidcLogoutEndpointController(final OidcConfigurationContext context,
+                                        OidcPostLogoutRedirectUrlMatcher postLogoutRedirectUrlMatcher) {
         super(context);
+        this.urlValidator = new UrlValidator(UrlValidator.ALLOW_LOCAL_URLS);
```

Comment:
Please obtain the logout validator from the context and do not instantiate it directly. There is a bean somewhere that does control the validator globally.

Reply:
the CAS UrlValidator unfortunately wraps the apache-commons UrlValidator and offers no possibility to configure the inner validator which does all the work - this is why I did it, localhost should (and must) be allowed in our case. Are you OK with me changing https://github.com/apereo/cas/blob/master/core/cas-server-core-web-api/src/main/java/org/apereo/cas/web/SimpleUrlValidator.java#L30 to allow also local Urls? Then I have no objections

Reply:
You should not have to. Here is the bean definition:

```java
    @Bean
    @RefreshScope
    public FactoryBean<UrlValidator> urlValidator() {
        val httpClient = this.casProperties.getHttpClient();
        val allowLocalLogoutUrls = httpClient.isAllowLocalUrls();
        val authorityValidationRegEx = httpClient.getAuthorityValidationRegex();
        val authorityValidationRegExCaseSensitive = httpClient.isAuthorityValidationRegExCaseSensitive();
        return new SimpleUrlValidatorFactoryBean(allowLocalLogoutUrls, authorityValidationRegEx, authorityValidationRegExCaseSensitive);
    }
```

So you would, I suppose:

1. Inject `urlValidator` into the component to you need.
2. set `cas.http-client.allow-local-urls=true`


Would that work?